File: data/src/main/java/dev/olog/data/api/lastfm/track/TrackInfo.java
package dev.olog.data.api.lastfm.track;

import androidx.annotation.Nullable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class TrackInfo {

    @Expose
    @Nullable
    public Track track;

    public static class Track {

        @Expose
        @Nullable
        public String name;

        @Expose
        @Nullable
        public String mbid;

        @Expose
        @Nullable
        public String listeners;

        @Expose
        @Nullable
        public Artist artist;

        @Expose
        @Nullable
        public Album album;

        @Expose
        @Nullable
        public Wiki wiki;

        public static class Image {

            @SerializedName("#text")
            @Expose
            @Nullable
            public String text;

            @Expose
            @Nullable
            public String size;
        }

        public class Album {

            @Expose
            @Nullable
            public String artist;
            @Expose
            @Nullable
            public String title;
            @Expose
            @Nullable
            public String mbid;
            @Expose
            @Nullable
            public String url;
            @Expose
            @Nullable
            public List<Image> image = new ArrayList<>();

        }

        public static class Artist {

            @Expose
            @Nullable
            public String name;
            @Expose
            @Nullable
            public String mbid;
            @Expose
            @Nullable
            public String url;
        }

        public static class Wiki {

            @Expose
            @Nullable
            public String summary;
            @Expose
            @Nullable
            public String content;

        }


    }

}

File: data/src/main/java/dev/olog/data/api/lastfm/track/TrackInfo.kt
package dev.olog.data.api.lastfm.track

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class TrackInfo(
    val track: Track? = null
) {

    @Serializable
    class Track(
        val name: String? = null,
        val mbid: String? = null,
        val listeners: String? = null,
        val artist: Artist? = null,
        val album: Album? = null,
        val wiki: Wiki? = null,
    )

    @Serializable
    class Image(
        @SerialName("#text")
        val text: String? = null,
        val size: String? = null,
    )

    @Serializable
    class Album(
        val artist: String? = null,
        val title: String? = null,
        val mbid: String? = null,
        val url: String? = null,
        val image: List<Image>? = null,
    )

    @Serializable
    class Artist(
        val name: String? = null,
        val mbid: String? = null,
        val url: String? = null,
    )

    @Serializable
    class Wiki(
        val summary: String? = null,
        val content: String? = null,
    )

}